app/controllers/line_items_controller.rb

```diff
@@ -20,21 +20,24 @@ def update_quantity
   end
 
   def update
-    ingredient = Ingredient.find_by(name: params[:line_item][:extra_ingredient])
-    if @line_item.ingredient_left(ingredient, @branch)
-      @line_item.update_columns(extra_ingredient: params[:line_item][:extra_ingredient])
-      redirect_with_flash("success", "extra_ingredient_added", store_index_url)
-    else
-      redirect_with_flash("danger", "insufficient_stock", store_index_url)
-    end
+    @line_item.update_columns(extra_ingredient: params[:line_item][:extra_ingredient])
```

Comment:
why are we not permitting the params here. Also handle case when line_items fails to update